src/broadcast.jl

```diff
@@ -8,199 +9,92 @@ else
-            @nexprs $nargs i->(keep_i = keeps[i])
-            @nexprs $nargs i->(Idefault_i = Idefaults[i])
-            if !lift
-                # destructure the keeps and As tuples
-                @nexprs $nargs i->(X_i = Xs[i])
-                @simd for I in CartesianRange(indices(Z))
-                    # reverse-broadcast the indices
-                    @nexprs $nargs i->(I_i = newindex(I, keep_i, Idefault_i))
-                    # extract array values
-                    @nexprs $nargs i->(@inbounds val_i = X_i[I_i])
-                    # call the function and store the result
-                    @inbounds Z[I] = @ncall $nargs f val
-                end
-            else
-                # destructure the indexmaps and Xs tuples
-                @nexprs $nargs i->(values_i = Xs[i].values)
-                @nexprs $nargs i->(isnull_i = Xs[i].isnull)
-                @simd for I in CartesianRange(indices(Z))
-                    # reverse-broadcast the indices
-                    @nexprs $nargs i->(I_i = newindex(I, keep_i, Idefault_i))
-                    if $nullcheck
-                        # if any args are null, store null
-                        @inbounds Z.isnull[I] = true
-                    else
-                        # extract array values
-                        @nexprs $nargs i->(@inbounds val_i = values_i[I_i])
-                        # call the function and store the result
-                        @inbounds Z[I] = @ncall $nargs f val
-                    end
-                end
-            end
-        end
-    end
+    using Base.Broadcast: ziptype
+end
 
-    """
-        broadcast!(f, B::NullableArray, As::NullableArray...; lift::Bool=false)
+invoke_broadcast!{F, N}(f::F, dest, As::Vararg{NullableArray, N}) =
+    invoke(broadcast!, Tuple{Function, AbstractArray, Vararg{AbstractArray, N}}, f, dest, As...)
 
-    This method implements the same behavior as that of `broadcast!` when called
-    on regular `Array` arguments. It also includes the `lift` keyword argument,
-    which when set to true will lift `f` over the entries of the `As`.
+"""
+    broadcast(f, As::NullableArray...)
 
-    Lifting is disabled by default. Note that this method's signature specifies
-    the destination `B` array as well as the source `As` arrays as all
-    `NullableArray`s. Thus, calling `broadcast!` on a arguments consisting of
-    both `Array`s and `NullableArray`s will fall back to the implementation of
-    `broadcast!` in `base/broadcast.jl`.
-    """
-    # Required to solve dispatch ambiguity between
-    #   broadcast!(f, X::AbstractArray, x::Number...)
-    #   broadcast!(f, Z::NullableArrays.NullableArray, Xs::NullableArrays.NullableArray...)
-    @inline Base.broadcast!(f, Z::NullableArray; lift=false) =
-        broadcast!(f, Z, Z; lift=lift)
+Call `broadcast` with nullable lifting semantics and return a `NullableArray`.
```

Comment:
Good catch. Actually, since in this PR for now `lift` isn't supposed to be overloaded (so that lifting is always done), the description is correct. I've removed line 62. I hope we can improve this later in another PR.